reporter/src/main/kotlin/reporters/ctrlx/CtrlXAutomationReporter.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters.ctrlx
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.licenses.LicenseView
+import org.ossreviewtoolkit.model.writeValue
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+
+class CtrlXAutomationReporter : Reporter {
+    companion object {
+        const val REPORT_FILENAME = "fossinfo.json"
+    }
+
+    override val reporterName = "CtrlXAutomation"
+
+    override fun generateReport(input: ReporterInput, outputDir: File, options: Map<String, String>): List<File> {
+        val reportFile = outputDir.resolve(REPORT_FILENAME)
+
+        val packages = input.ortResult.getPackages(omitExcluded = true)
+        val components = packages.mapTo(mutableListOf()) { (pkg, _) ->
+            val qualifiedName = when (pkg.id.type) {
+                "NPM" -> with(pkg.id) {
+                    listOfNotNull(namespace.takeIf { it.isNotEmpty() }, name).joinToString("/")
+                }
+                else -> pkg.id.name
```

Comment:
Go packages don't require a special handling for the qualifiedName?

Reply:
I don't know, I only have an example for NPM packages. I've added a comment to explain that further additions might be needed for other package managers.